docs/data/material/experimental-api/css-variables/css-variables.md

```diff
@@ -0,0 +1,219 @@
+
+Play around with the demo below!
+
+{{"demo": "CssVariablesCustomization.js", "iframe": true }}
+
+If you are using TypeScript you should use module augmentation to update the `Theme` structure:
+
+```tsx
+import { Theme as MuiTheme } from '@mui/material/styles';
+
+declare module '@mui/material/styles' {
+  interface Theme {
+    vars: Omit<
+      MuiTheme,
+      'typography' | 'mixins' | 'breakpoints' | 'direction' | 'transitions'
+    >;
+  }
+}
+```
+
+### Customizing the theme
+
+If you want, for example, to override Material UI's default color schemes, you can use the `experimental_extendTheme` utility.
+
+```jsx
+const theme = experimental_extendTheme({
+  colorSchemes: {
+    light: {
+      palette: {
+        primary: teal,
+        secondary: deepOrange,
+      },
+    },
+    dark: {
+      palette: {
+        primary: cyan,
+        secondary: orange,
+      },
+    },
+  },
+});
+```
+
+{{"demo": "CssVarsCustomTheme.js", "iframe": true }}
+
+If you are using [`ThemeProvider`](/material-ui/customization/theming/#theme-provider), you can replace it with the new experimental provider.
+
+```diff
+- import { ThemeProvider, createTheme } from '@mui/material/styles';
++ import { Experimental_CssVarsProvider as CssVarsProvider } from '@mui/material/styles';
+
+function App() {
+  return (
+-    <ThemeProvider theme={createTheme()}>
+-      ...
+-    </ThemeProvider>
++    <CssVarsProvider>
++      ...
++    </CssVarsProvider>
+  )
+}
+```
+
+### Toggle between light and dark mode
+
+`Experimental_CssVarsProvider` provides light and dark mode by default.
+It stores the selected user mode and syncs it with the browser's local storage internally.
+You can read and update the mode via the `useColorScheme` API.
+
+```jsx
+import {
+  Experimental_CssVarsProvider as CssVarsProvider,
+  useColorScheme,
+} from '@mui/material/styles';
+
+const ModeSwitcher = () => {
+  const { mode, setMode } = useColorScheme();
+  const [mounted, setMounted] = React.useState(false);
+
+  React.useEffect(() => {
+    setMounted(true);
+  }, []);
+
+  if (!mounted) {
+    // for server-side rendering
+    // Read more on https://github.com/pacocoursey/next-themes#avoid-hydration-mismatch
+    return null;
+  }
+
+  return (
+    <Button
+      variant="outlined"
+      onClick={() => {
+        if (mode === 'light') {
+          setMode('dark');
+        } else {
+          setMode('light');
+        }
+      }}
+    >
+      {mode === 'light' ? 'Dark' : 'Light'}
+    </Button>
+  );
+};
+
+function App() {
+  return (
+    <CssVarsProvider>
+      <ModeSwitcher />
+    </CssVarsProvider>
+  );
+}
+```
+
+### Server-side rendering
+
+To prevent the dark-mode SSR flickering during the hydration phase, place `getInitColorSchemeScript()` before the `<Main />` tag.
+
+#### Next.js
+
+To use the API with a Next.js project, add the following code to the `pages/_document.js` file.
```

Comment:
Colon. Also, I think it's worth adding a reference for the custom `_document` since this is an advanced feature (and some users might be confused when they can't find it in their fresh app).

```suggestion
To use the API with a Next.js project, add the following code to the custom [`pages/_document.js`](https://nextjs.org/docs/advanced-features/custom-document) file:
```